### ISSUE TITLE
Can't pass React component as placeholder anymore

### ISSUE THREAD

**Issue by czechdave** [ianstormtaylor/slate#2977]
#### Do you want to request a _feature_ or report a _bug_?
BUG

#### What's the current behavior?
`placeholder` prop on `<Editor />` can't be a React component anymore. If it's not a `string` the editor crashes.

#### What's the expected behavior?
The `<Editor />` just renders `placeholder` if it's a React component.

https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react-placeholder/src/index.js#L22

**Comment by kilrain** [comment]
I brought this up in the slack channel recently: https://slate-js.slack.com/archives/C1RH7AXSS/p1564944468254200

**Comment by gracicot** [comment]
To workaround this issue we decided to re-implement the placeholder plugin. We needed the ability to render a react element because our placeholder was allowing text wrapping (we have a whole welcome message in it).

To allow the text wrapping we used absolute positionning and a relative element, which both have different styling properties. Doing that without react elements is not possible as far as I know.
